typed/merge.go

```diff
@@ -288,31 +310,33 @@ func (w *mergingWalker) visitMapItems(t schema.Map, lhs, rhs *value.Map) (errs V
 			if newErrs := w2.merge(); len(newErrs) > 0 {
 				errs = append(errs, newErrs...)
 			} else if w2.out != nil {
-				out.Set(name, *w2.out)
+				out.Items = append(out.Items, value.Field{litem.Name, *w2.out})
```

Comment:
I'm curious about that, isn't it explicitely against the doc for `Map`?

Reply:
It's OK here because
* out is brand new here
* Get isn't called until after all items have been added
* we know we won't add duplicate items